client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you-plugin-section.tsx

```diff
@@ -79,6 +86,32 @@ export const ThankYouPluginSection = ( { plugin }: { plugin: any } ) => {
 		plugin?.setup_url && siteAdminUrl ? siteAdminUrl + plugin.setup_url : null;
 	const setupURL = plugin?.action_links?.Settings || fallbackSetupUrl || managePluginsUrl;
 	const documentationURL = plugin?.documentation_url;
+	const purchases = useSelector( ( state ) => getUserPurchases( state ) );
+	const isLoadingPurchases = useSelector(
+		( state ) => isFetchingUserPurchases( state ) || ! hasLoadedUserPurchasesFromServer( state )
+	);
+	useQueryUserPurchases();
+
+	const productPurchase = purchases?.find( ( item ) => {
+		if ( item.siteId === siteId ) {
+			if (
+				plugin.variations?.monthly?.product_id === item.productId ||
+				plugin.variations?.yearly?.product_id === item.productId
+			) {
+				return item;
+			}
+		}
+	} );
+
+	if ( ! isLoadingPurchases ) {
+		if ( productPurchase ) {
+			plugin.expirationDate = translate( 'Expires on %s', {
```

Comment:
A bit different from Figma as [I'm reusing this string](https://github.com/Automattic/wp-calypso/blob/trunk/client/components/product-expiration/index.tsx#L53).